src/common/com/intellij/plugins/haxe/lang/psi/impl/HaxeNamedElementImpl.java

```diff
@@ -57,8 +57,9 @@ public PsiElement setName(@NonNls @NotNull String newElementName) throws Incorre
       getNode().replaceChild(identifier.getNode(), identifierNew.getNode());
     }
 
-    if (getParent() instanceof HaxeClass) {
-      final HaxeFile haxeFile = (HaxeFile)getParent().getParent();
+    PsiElement parent = getParent();
+    if (parent instanceof HaxeClass && parent.getParent() instanceof HaxeClass) {
+      final HaxeFile haxeFile = (HaxeFile)parent.getParent();
```

Comment:
What are you trying to check here?  That parent.getParent() is actually a HaxeClass, or that it's a HaxeFile?  (Seems the latter is what is intended.)